exist-core/src/main/resources/org/exist/xquery/lib/xqsuite/xqsuite.xql

```diff
@@ -706,13 +795,13 @@ declare %private function test:assertEquals($values as item()*, $result as item(
  : Check for equality of the function's result with the value in the annotation.
  : This function transforms the result to a string before checking for equality.
  :)
-declare %private function test:assertEqualsPermutation($values as item()*, $result as item()*) as element(report)? {
-    if (exists($values)) then
-        if (count($values) eq count($result)) then
+declare %private function test:assertEqualsPermutation($values as element(value)*, $result as item()*) as element(report)? {
+    if (fn:not(fn:empty($values))) then
```

Comment:
why is `exists()` not sufficient here?

Reply:
I changed it back. There was a mix of both styles already present in `xqsuite.xql` before even my changes